05-Mining-Protocol.md

```diff
@@ -99,15 +99,17 @@ Figure 4.3.b Mining Protocol Messages: Mining on Standard Channel
 
-| Field Name               | Bit | Description                                                                         |
-| ------------------------ | --- | ----------------------------------------------------------------------------------- |
-| REQUIRES_STANDARD_JOBS   | 0   | The downstream node requires standard jobs. It does not understand group channels - |
-|                          |     | it is unable to process extended jobs sent to standard channels through a group     |
-|                          |     | channel.                                                                            |
-| REQUIRES_WORK_SELECTION  | 1   | If set to 1, the client notifies the server that it will send SetCustomMiningJob on |
-|                          |     | this connection                                                                     |
-| REQUIRES_VERSION_ROLLING | 2   | The client requires version rolling for efficiency or correct operation and the     |
-|                          |     | server MUST NOT send jobs which do not allow version rolling                        |
+| Field Name                              | Bit | Description                                                                                                                      |
+|-----------------------------------------|-----|----------------------------------------------------------------------------------------------------------------------------------|
+| REQUIRES_STANDARD_JOBS                  | 0   | The downstream node requires standard jobs. It does not understand group channels -                                              |
+|                                         |     | it is unable to process extended jobs sent to standard channels through a group                                                  |
+|                                         |     | channel.                                                                                                                         |
+| REQUIRES_WORK_SELECTION                 | 1   | If set to 1, the client notifies the server that it will send SetCustomMiningJob on                                              |
+|                                         |     | this connection                                                                                                                  |
+| REQUIRES_VERSION_ROLLING                | 2   | The client requires version rolling for efficiency or correct operation and the                                                  |
+|                                         |     | server MUST NOT send jobs which do not allow version rolling                                                                     |
+| REQUIRES_EXTENDED_SHARES_IDENTIFICATION | 3   | If set, the client requests that the server support the inclusion of a user_identity field in the SubmitSharesExtended message.  |
```

Comment:
This is strange.

The client is the one identifying the shares. So it's not true. The client doesn't require anything. It only tells the server it's going to identify the shares. But that's not the whole story.
What it really tells is that it's going to use a new message format with an additional field. So it in fact does what the version field is supposed to do - negotiates protocol version.

Or what else does it negotiate?
What if it sets this flag and then doesn't identify the shares? Should those shares be rejected? If not then what's the point? Or of it doesn't set the flag and then the shares are identified? That's ok? Again, what's the point?